src/main/java/com/unascribed/fabrication/mixin/g_weird_tweaks/villagers_follow_emerald_blocks/MixinVillagerEntity.java

```diff
@@ -0,0 +1,47 @@
+package com.unascribed.fabrication.mixin.g_weird_tweaks.villagers_follow_emerald_blocks;
+
+import com.unascribed.fabrication.support.EligibleIf;
+import com.unascribed.fabrication.support.MixinConfigPlugin;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.ai.TargetPredicate;
+import net.minecraft.entity.passive.MerchantEntity;
+import net.minecraft.entity.passive.VillagerEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Items;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import java.util.stream.StreamSupport;
+
+@Mixin(VillagerEntity.class)
+@EligibleIf(configEnabled="*.villagers_follow_emerald_blocks")
+public abstract class MixinVillagerEntity extends MerchantEntity  {
+	PlayerEntity player = null;
```

Comment:
This name is too generic, and the field should be `private`.

```suggestion
	private PlayerEntity fabrication$player = null;
```